app/code/Magento/StoreGraphQl/Model/Resolver/Store/StoreConfigDataProvider.php

```diff
@@ -55,24 +56,26 @@ public function __construct(
      */
     public function getStoreConfigData(StoreInterface $store): array
     {
-        $storeConfigData = array_merge(
-            $this->getBaseConfigData($store),
-            $this->getExtendedConfigData((int)$store->getId())
-        );
-        return $storeConfigData;
+        $stores = $this->storeConfigManager->getStoreConfigs();
+        $defaultStoreConfig = current($this->storeConfigManager->getStoreConfigs([$store->getCode()]));
```

Comment:
`\Magento\Store\Model\Service\StoreConfigManager` does not cache result of `getStoreConfigs`.
Maybe we can return a list of configs and work with the list to avoid extra queries to DB.

Reply:
Hello @swnsma , the change has been done. Please let me know if any questions.